### PR TITLE
OSDOCS#6633 - Add NFS support for Azure File CSI driver

### PR DESCRIPTION
**Pull request by lpettyjo**
Version(s): 4.14+

Issue: https://issues.redhat.com/browse/OSDOCS-6633

Rel Note Known issue addition added to JIRA ticket: https://issues.redhat.com/browse/OCPBUGS-18581

Link to docs preview: https://61831--docspreview.netlify.app/openshift-enterprise/latest/storage/container_storage_interface/persistent-storage-csi-azure-file.html#persistent-storage-csi-azure-file-nfs_persistent-storage-csi-azure-file

QE review:
- [ ] QE has approved this change.

Additional information:

**PTAL**: @bertinatto, @duanwei33, @gcharot

### CODE REVIEW

File: modules/persistent-storage-csi-azure-file-nfs.adoc [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+// * storage/container_storage_interface/persistent_storage-csi-azure-file.adoc
+//
+
+:_content-type: CONCEPT
+[id="persistent-storage-csi-azure-file-nfs_{context}"]
+= NFS support
+
+{product-title} 4.14, and later, supports Azure File Container Storage Interface (CSI) Driver Operator with  Network File System (NFS) with the following caveats:
+
+* Azure File with NFS applies a FS group policy at the driver level and pods cannot override it.
```
Comment (gcharot) [inline modules/persistent-storage-csi-azure-file-nfs.adoc:12]: I would say something such as:
Azure File CSI with NFS does not honor the fsGroupChangePolicy requested by pods. Azure File CSI with NFS applies a default OnRootMismatch fsGroup policy regardless of the policy requested by the pod.

I wonder if we shouldn't put that statement inside an "important" red square.
cc @gnufied @bertinatto

File: modules/persistent-storage-csi-azure-file-nfs.adoc [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+* The Azure File CSI Operator does not automatically create a storage class for NFS. You need to create it manually.
```
Comment (gcharot) [inline modules/persistent-storage-csi-azure-file-nfs.adoc:14]: We need to add a snippet of the storage class as per our testing.
cc @gnufied @bertinatto @duanwei33

Comment (JoeAldinger) [inline modules/persistent-storage-csi-azure-file-nfs.adoc:14]:
```suggestion
* The Azure File CSI Operator does not automatically create a storage class for NFS. You must create it manually.
```

File: modules/persistent-storage-csi-azure-file-nfs.adoc [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// Module included in the following assemblies:
+//
+// * storage/container_storage_interface/persistent_storage-csi-azure-file.adoc
+//
+
+:_content-type: CONCEPT
+[id="persistent-storage-csi-azure-file-nfs_{context}"]
+= NFS support
```
Comment (JoeAldinger) [inline modules/persistent-storage-csi-azure-file-nfs.adoc:8]:
```suggestion
= Network File System support
```

Comment (JoeAldinger) [reply modules/persistent-storage-csi-azure-file-nfs.adoc:8]: This is the first time that NFS is mentioned in this assembly so I'd recommend writing it out.

File: modules/persistent-storage-csi-azure-file-nfs.adoc (hunk 1 above)
Comment (JoeAldinger) [inline modules/persistent-storage-csi-azure-file-nfs.adoc:12]:
```suggestion
* Azure File with NFS applies a File System (FS) group policy at the driver level and pods cannot override it.
```

**Review by JoeAldinger — COMMENTED** [review]
A few small suggestions.

File: modules/persistent-storage-csi-azure-file-nfs.adoc (hunk 2 above)
Comment (gnufied) [reply modules/persistent-storage-csi-azure-file-nfs.adoc:14]: Yeah so something like:

```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: azurefile-csi-nfs
provisioner: file.csi.azure.com
parameters:
  protocol: nfs
  skuName: Premium_LRS  # available values: Premium_LRS, Premium_ZRS
mountOptions:
  - nconnect=4
 ```

File: modules/persistent-storage-csi-azure-file-nfs.adoc (hunk 1 above)
Comment (gnufied) [reply modules/persistent-storage-csi-azure-file-nfs.adoc:12]: +1 to updated wording.

Comment (lpettyjo) [reply modules/persistent-storage-csi-azure-file-nfs.adoc:12]: Fixed.

File: modules/persistent-storage-csi-azure-file-nfs.adoc (hunk 2 above)
Comment (lpettyjo) [reply modules/persistent-storage-csi-azure-file-nfs.adoc:14]: Done.

**Comment by duanwei33** [comment]
Comment on something which in not included in this PR but related to:

```
Azure File CSI Driver Operator does not support:
Virtual hard disks (VHD)
Running on nodes with FIPS mode enabled.
```

We should update that SMB file share doesn't support running on nodes with FIPS mode enabled because NFS protocol supports and we are adding it.

cc @gcharot @bertinatto @gnufied

**Comment by bertinatto** [comment]
> Comment on something which in not included in this PR but related to:
> 
> ```
> Azure File CSI Driver Operator does not support:
> Virtual hard disks (VHD)
> Running on nodes with FIPS mode enabled.
> ```
> 
> We should update that SMB file share doesn't support running on nodes with FIPS mode enabled because NFS protocol supports and we are adding it.
> 
> cc @gcharot @bertinatto @gnufied

Good point. SMB doesn't work in FIPS mode, but NFS does.

**Comment by gcharot** [comment]
Yes good point, we should document the FIPS part. thanks!

**Comment by duanwei33** [comment]
LGTM

**Comment by gcharot** [comment]
LGTM

**Comment by gcharot** [comment]
LGTM

**Review by bertinatto — COMMENTED** [review]
LGTM

**Comment by lpettyjo** [comment]
On hold due to https://issues.redhat.com/browse/OCPBUGS-18581.

**Comment by lpettyjo** [comment]
Changes for On hold due to https://issues.redhat.com/browse/OCPBUGS-18581 have been implemented.

**Comment by lpettyjo** [comment]
Need LGTM for new changes for https://issues.redhat.com/browse/OCPBUGS-18581 issue from @bertinatto, @duanwei33,  @gcharot.

**Comment by duanwei33** [comment]
LGTM
I'm okay to keep only one simple WA for [OCPBUGS-18581](https://issues.redhat.com/browse/OCPBUGS-18581)

**Comment by gcharot** [comment]
LGTM

**Comment by bertinatto** [comment]
lgtm

**Pull request merged**